File: src/main/frontend/src/features/dashboard/domain/MatchSummary.ts
export interface MatchSummary {
  matchId: string;
  gameDuration: number;
  gameCreation: number;
  queueId: number;
  win: boolean;
  championId: number;
  championName: string;
  kills: number;
  deaths: number;
  assists: number;
  totalMinionsKilled: number;
  neutralMinionsKilled: number;
}

File: src/main/java/com/analysisgg/modules/riotapi/domain/model/MatchSummary.java
package com.analysisgg.modules.riotapi.domain.model;

import java.util.List;

public record MatchSummary(
    String matchId,
    long gameDuration,
    long gameCreation,
    int queueId,
    boolean win,
    int championId,
    String championName,
    int kills,
    int deaths,
    int assists,
    int totalMinionsKilled,
    int neutralMinionsKilled,
    String teamPosition,
    List<ParticipantSummary> participants
) {
}
